### PR TITLE
Fix annotate for chunked operations like in insulation

### PR DESCRIPTION
**Pull request by Phlya**
Simple solution with .loc instead of .iloc - probaly a bit slower, perhaps some arithmetic can fix the .iloc to behave correctly

### CODE REVIEW

**Comment by Phlya** [comment]
I had to add the recently removed export of _IndexingMixin, since cooltools uses it...

**Comment by Phlya** [comment]
@nvictus would you be able to review? We should release a fix asap

**Comment by nvictus** [comment]
This needs a test case that triggers the error without the fix.

File: src/cooler/api.py [hunk 1]
```diff
@@ -662,6 +661,16 @@ def _loc_slice(df, beg, end):
...
+    # Drop original bin IDs if not wanted
+    if replace:
+        cols_to_drop = [col for col in ("bin1_id", "bin2_id") if col in columns]
+        pixels = pixels.drop(cols_to_drop, axis=1)
+
+    # Concatenate bin annotations with pixels
+    out = pd.concat([*anns, pixels.reset_index(drop=True)], axis=1)
+    out.index = pixels.index
+    return out
+
```
Comment (nvictus) [inline src/cooler/api.py:673]: This looks like an unintentional duplication of the lines above.

**Comment by Phlya** [comment]
Thank you @nvictus!
For now I am keeping the _IndexingMixin, since without it cooltools just breaks completely... We can migrate it in the next cooltools release and then remove from cooler.
Added a test, it fails in the master branch but works in the version here in my local tests.

Of course numpy 2.0 breaks stuff, so the tests here fail...

**Comment by nvictus** [comment]
Great opportunity to throw in the numpy major version pin.

**Comment by Phlya** [comment]
Looks like disallowing numpy 2.0 makes the tests pass! Shall we merge and make a bugfix release?

**Pull request merged**